services/sky/document_view.h

```diff
@@ -41,14 +35,13 @@ class TextureLayer;
 
 class DocumentView : public blink::ServiceProvider,
                      public blink::SkyViewClient,
-                     public mojo::ViewManagerDelegate,
-                     public mojo::ViewObserver,
+                     public mojo::NativeViewportEventDispatcher,
                      public sky::LayerClient,
                      public sky::LayerHostClient,
                      public ui::GestureConsumer {
  public:
-  DocumentView(mojo::InterfaceRequest<mojo::ServiceProvider> services,
-               mojo::ServiceProviderPtr exported_services,
+  DocumentView(mojo::InterfaceRequest<mojo::ServiceProvider> exported_services,
+               mojo::ServiceProviderPtr imported_services,
```

Comment:
Yes you're right.  I misspoke.  These do come from the connection in which
case I do think it makes sense for a Sky app to be able to receive or
provide services to whatever is connecting to it.  So these are necessary.

As for the names, these are the ones used downstream in the Sky framework
and it makes sense to adapt them at this layer.  I'm actually inclined to
change the Application.mojom to use these names in place of "services" and
"exposed_services" as it does today which is less precise.

On Wed, Sep 23, 2015 at 4:03 PM James Robinson <notifications@github.com>
wrote:

> In services/sky/document_view.h
> <https://github.com/flutter/engine/pull/1276#discussion_r40269114>:
>
> >                       public sky::LayerClient,
> >                       public sky::LayerHostClient,
> >                       public ui::GestureConsumer {
> >   public:
> > -  DocumentView(mojo::InterfaceRequest<mojo::ServiceProvider> services,
> > -               mojo::ServiceProviderPtr exported_services,
> > +  DocumentView(mojo::InterfaceRequest<mojo::ServiceProvider> exported_services,
> > +               mojo::ServiceProviderPtr imported_services,
>
> These are the serviceprovider pipes particular to a connection. This
> connection will be from the shell (since it's a primordial app) but that
> doesn't mean that it's the thing to request services from the shell - the
> way to do that is to call Shell::ConnectToApplication(). So I think you
> should drop these. If there's something I'm missing with this then please
> make the names match the mojom exactly.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/flutter/engine/pull/1276/files#r40269114>.
>